test/e2e/kubevirt.go

```diff
@@ -2521,4 +2563,62 @@ ethernets:
 			)), "second VM should not be ready due to MAC conflict")
 		})
 	})
+
+	Context("subnet exhaustion", func() {
+		const networkName = "net1"
+		var (
+			cudn     *udnv1.ClusterUserDefinedNetwork
+			cidrIPv4 = "10.130.0.0/30" // no usable IPs (.1 and .2, .0 is network infra owned, .3 is broadcast)
+			cidrIPv6 = "2011:100:200::0/120"
+		)
+
+		BeforeEach(func() {
+			l := map[string]string{
+				"e2e-framework":           fr.BaseName,
+				RequiredUDNNamespaceLabel: "",
+			}
+			ns, err := fr.CreateNamespace(context.TODO(), fr.BaseName, l)
+			Expect(err).NotTo(HaveOccurred())
+			fr.Namespace = ns
+			namespace = fr.Namespace.Name
+
+			dualCIDRs := filterDualStackCIDRs(fr.ClientSet, []udnv1.CIDR{udnv1.CIDR(cidrIPv4), udnv1.CIDR(cidrIPv6)})
+			cudn, _ = kubevirt.GenerateCUDN(namespace, networkName, udnv1.NetworkTopologyLayer2, udnv1.NetworkRolePrimary, dualCIDRs)
+			createCUDN(cudn)
+		})
+
+		It("should fail when subnet is exhausted", func() {
+			By("Creating VM that should fail due to subnet exhaustion")
+			exhaustedVMName := "exhausted-vm"
+			vm := fedoraWithTestToolingVM(
+				nil, // labels
+				nil, // no static IP annotations
+				nil, // nodeSelector
+				kubevirtv1.NetworkSource{
+					Pod: &kubevirtv1.PodNetwork{},
+				},
+				`#cloud-config
+password: fedora
+chpasswd: { expire: False }
+`,
+				`version: 2
+ethernets:
+  eth0:
+    dhcp4: true
+    dhcp6: true
```

Comment:
I don't think this will work; I think that you need to template this properly depending on the results from `filterDualStackCIDRs` . i.e. if the cluster is:
- single stack ipv4: only enable dhcp4
- single stack ipv6: only enable dhcp6
- dual stack: enable both, as you have now

@qinqon to keep me honest. I recall some issues when you enabled the "wrong" family and there was no way to get an IP from it.

Reply:
So why don't we do it on any of the current tests?